httpie/core.py

```diff
@@ -143,50 +147,64 @@ def program(args, env, log_error):
+                    if is_py3:
+                        has_branched('program', 9)
+                        if 'colors' in args.prettify:
+                            has_branched('program', 10)
+                            write_stream_with_colors_win_py3(**write_stream_kwargs)
                 else:
-                    has_branched('program', 9)
+                    has_branched('program', 11)
                     write_stream(**write_stream_kwargs)
             except IOError as e:
-                has_branched('program', 10)
-                if not show_traceback and e.errno == errno.EPIPE:
-                    has_branched('program', 11)
-                    # Ignore broken pipes unless --traceback.
-                    env.stderr.write('\n')
+                has_branched('program', 12)
+                if not show_traceback:
+                    has_branched('program', 13)
+                    if e.errno == errno.EPIPE:
+                        has_branched('program', 14)
+                        # Ignore broken pipes unless --traceback.
+                        env.stderr.write('\n')
                 else:
-                    has_branched('program', 12)
+                    has_branched('program', 15)
                     raise
 
-        if downloader and exit_status == ExitStatus.OK:
-            has_branched('program', 13)
-            # Last response body download.
-            download_stream, download_to = downloader.start(final_response)
-            write_stream(
-                stream=download_stream,
-                outfile=download_to,
-                flush=False,
-            )
-            downloader.finish()
-            if downloader.interrupted:
-                has_branched('program', 14)
-                exit_status = ExitStatus.ERROR
-                log_error('Incomplete download: size=%d; downloaded=%d' % (
-                    downloader.status.total_size,
-                    downloader.status.downloaded
-                ))
+        if downloader:
+            has_branched('program', 16)
+            if exit_status == ExitStatus.OK:
+                has_branched('program', 17)
+                # Last response body download.
+                download_stream, download_to = downloader.start(final_response)
+                write_stream(
+                    stream=download_stream,
+                    outfile=download_to,
+                    flush=False,
+                )
+                downloader.finish()
+                if downloader.interrupted:
+                    has_branched('program', 18)
+                    exit_status = ExitStatus.ERROR
+                    log_error('Incomplete download: size=%d; downloaded=%d' % (
+                        downloader.status.total_size,
+                        downloader.status.downloaded
+                    ))
         return exit_status
 
     finally:
         if downloader and not downloader.finished:
```

Comment:
This should be modified to "if downloader" right?

Reply:
yes